src/schema/posts.ts

```diff
@@ -1101,6 +1224,56 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
   BaseContext
 >({
   Query: {
+    sourcePostModerationsById: async (
+      _,
+      { sourceId }: { sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSourcePostModeration[]> => {
+      const isModerator = await isPrivilegedMember(ctx, sourceId);
+      if (isModerator) {
+        return graphorm.query<GQLSourcePostModeration>(
+          ctx,
+          info,
+          (builder) => ({
+            ...builder,
+            queryBuilder: builder.queryBuilder.where(
+              `"${builder.alias}"."sourceId" = :sourceId`,
+              { sourceId },
+            ),
+          }),
+        );
+      }
+      const { userId } = ctx;
+      return graphorm.query<GQLSourcePostModeration>(ctx, info, (builder) => ({
+        ...builder,
+        queryBuilder: builder.queryBuilder.where(
+          `"${builder.alias}"."sourceId" = :sourceId AND "${builder.alias}"."createdById" = :userId`,
+          { sourceId, userId },
+        ),
+      }));
+    },
+    sourcePostModeration: async (
+      _,
+      { id, sourceId }: { id: string; sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      const isModerator = await isPrivilegedMember(ctx, sourceId);
+      if (!isModerator) throw new ForbiddenError('Access denied!');
```

Comment:
@AmarTrebinjac for this one though `ensureSourcePermission` is exactly what we want no?